packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/notification/NotificationController.java

```diff
@@ -0,0 +1,65 @@
+package software.uncharted.terarium.hmiserver.controller.notification;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.ArraySchema;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import software.uncharted.terarium.hmiserver.models.notification.NotificationGroup;
+import software.uncharted.terarium.hmiserver.security.Roles;
+import software.uncharted.terarium.hmiserver.service.CurrentUserService;
+import software.uncharted.terarium.hmiserver.service.notification.NotificationService;
+
+@RequestMapping("/notification")
+@RestController
+@Slf4j
+@RequiredArgsConstructor
+public class NotificationController {
+
+    private final CurrentUserService currentUserService;
+    private final NotificationService notificationService;
+
+    @GetMapping("/user")
+    @Secured(Roles.USER)
+    @Operation(summary = "Return all recent notification groups for a user")
+    @ApiResponses(
+            value = {
+                @ApiResponse(
+                        responseCode = "200",
+                        description = "Returned recent notifications successfully",
+                        content =
+                                @Content(
+                                        array =
+                                                @ArraySchema(
+                                                        schema =
+                                                                @io.swagger.v3.oas.annotations.media.Schema(
+                                                                        implementation = NotificationGroup.class)))),
+                @ApiResponse(
+                        responseCode = "500",
+                        description = "There was an issue fetching the notifications",
+                        content = @Content)
+            })
+    public ResponseEntity<List<NotificationGroup>> getNotificationGroups(
```

Comment:
Can we have an option to limit the number of notification events returned for a notification group? I think most of the cases, we need the latest notification event per group and maybe we need to fetch more when needed or requested by user for the history.

Reply:
That's a great point, and kind of what I was asking above, maybe we need an option to do that.
Some Notification will need to keep every event, but most don't

Reply:
So it turns out, hibernate does not let you filter join fields dynamically. And because hibernate is infuriating, I can't even filter them out in the java code without it persisting those changes without disabling all of join field functionality altogether.

Reply:
So what would be a solution here?